scene/2d/cpu_particles_2d.cpp

```diff
@@ -1211,6 +1215,27 @@ void CPUParticles2D::_notification(int p_what) {
 			}
 
 			RS::get_singleton()->canvas_item_add_multimesh(get_canvas_item(), multimesh, texrid);
+
+#ifdef TOOLS_ENABLED
+			Vector2 gizmo_position = Vector2();
+			if (!local_coords) {
+				gizmo_position = get_position();
+			}
+
+			switch (emission_shape) {
+				case CPUParticles2D::EMISSION_SHAPE_RECTANGLE:
+					draw_rect(Rect2(gizmo_position - emission_rect_extents, emission_rect_extents * 2.0), Color(0.8, 0.7, 0.4, 0.4), false);
+					break;
+				case CPUParticles2D::EMISSION_SHAPE_SPHERE:
+					[[fallthrough]];
+				case CPUParticles2D::EMISSION_SHAPE_SPHERE_SURFACE:
+					draw_circle(gizmo_position, emission_sphere_radius, Color(0.8, 0.7, 0.4, 0.4), false);
+					break;
+				default:
```

Comment:
You could add a case for `EMISSION_SHAPE_POINTS` and `EMISSION_SHAPE_DIRECTED_POINTS` too, it's not difficult.

Reply:
Could you clarify this a bit? The points are written to an array style texture which I'm not sure how to display effectively every frame.

Reply:
In CPU Particles the emission points are a simple array of vectors.
For GPU Particles it's more complicated, yeah.

Reply:
Ah got you, thanks

Reply:
I investigated adding this, but I don't think it makes sense. Even in the CPU particles, the points are a list of points that particles will randomly spawn at, not points that make up a shape that particles will spawn within. I don't see a nice way of rendering hundreds of points in the gizmo rendering that is readable for the user.